lib/board.rb

```diff
@@ -0,0 +1,69 @@
+class Board
+
+  attr_accessor :cells
+
+
+  def initialize
+    self.cells ||= Array.new(9, " ")
+  end # initialize
+
+
+  ## displays the board
+  def display
+
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+
+  end # board
+
+
+  ## resets the array if necessary
+  def reset!
+    self.cells = Array.new(9, " ")
+  end # reset!
+
+
+  ## determines the value in the input position
+  def position(user_input)
+    cells[user_input.to_i - 1]
+  end # position
+
+
+  ## determines if the board is full
+  def full?
+    true if cells.all?{|pos| pos == "X" || pos == "O"}
+  end # full?
+
+
+  ## determines what turn it is now
+  def turn_count
+    count = 0
+    cells.each {|pos| count += 1 if pos == "X" || pos == "O"}
+    count
+  end # turn_count
+
+
+  ## determines if the spot has been taken
+  def taken?(user_input)
+    position(user_input) == "X" || position(user_input) == "O" ? true : false
```

Comment:
Avoid using the ternary operator to return `true` and `false`. Instead, rely on the implicit boolean return value of the expression. 

`cells.none? {|cell| cell == " " || cell == nil}`